src/lib/address_resolve/AddressResolve_DefaultImpl.h

```diff
@@ -203,6 +216,22 @@ class Resolver : public ::chip::AddressResolve::Resolver, public Dnssd::Operatio
     System::Layer * mSystemLayer = nullptr;
     Time::TimeSource<Time::Source::kSystem> mTimeSource;
     IntrusiveList<NodeLookupHandle> mActiveLookups;
+
+#if CHIP_DEVICE_ENABLE_CASE_DNS_CACHE
+    // Cache entry structure
+    struct CacheEntry
```

Comment:
I think we should separate the cache in a separate class whose only purpose is to maintain a cache and can be unit tested.

Also please add unit test. Note that https://github.com/project-chip/connectedhomeip/blob/master/src/lib/address_resolve/tests/TestAddressResolve_DefaultImpl.cpp already exists.

Reply:
moved to a separate class with unit tests